### PR TITLE
Add the missing i386 slice to the Geth compiled library.

### PR DESCRIPTION
**Pull request by Elaz**
This adds the i386 slice for iOS Geth framework.

### CODE REVIEW

File: build/ci.go [hunk 1]
```diff
@@ -839,6 +839,8 @@ func doXCodeFramework(cmdline []string) {
 	env := build.Env()
 
 	// Build the iOS XCode framework
+	build.MustRun(goTool("get", "-d", "golang.org/x/mobile/cmd/gomobile"))
```
Comment (OrenZak) [inline build/ci.go:842]: do we really need "-d" option? and if we do, I guess we don't need the line number 844 (duplicated).

**Comment by OrenZak** [comment]
Can you explain what was the problem?

File: build/ci.go (hunk 1 above)
Comment (Elaz) [reply build/ci.go:842]: Oh we definitely need 844, it's deliberate. -d tells go to just download and not compile anything. 844 will actually compile it without downloading.

**Comment by Elaz** [comment]
The problem was that geth isn't compiled for i386. For some iOS projects, it means supporting armv7 devices is limited in release builds.

**Comment by bolshoy** [comment]
@OrenZak when you run `go get`, it will download, compile and install the resulting libraries. But we need to patch the code after it's downloaded but before it's compiled. So the trick is to supply `-d` to `go get` so it will just download the code and then we patch it. When `go get` is run again, it finds the (patched) source locally and just compiles and installs it.

**Comment by OrenZak** [comment]
@bolshoy  yup understood 👍

**Pull request closed without merging**